projects/plugins/jetpack/extensions/shared/external-media/media-category/index.ts

```diff
@@ -0,0 +1,203 @@
+}
+
+/**
+ * External media sources.
+ */
+// eslint-disable-next-line no-shadow
+enum MediaSource {
+	Pexels = 'pexels',
+	GooglePhotos = 'google_photos',
+}
+
+/**
+ * Gutenberg media category search DTO.
+ */
+type MediaSearch = {
+	per_page: number;
+	search: string;
+};
+
+/**
+ * Gutenberg media item DTO.
+ */
+type MediaItem = {
+	sourceId: string;
+	id: string;
+	caption: string;
+	previewUrl: string;
+	title: string;
+	url: string;
+};
+
+/**
+ * WPCOM media list item DTO.
+ */
+type WpcomMediaItem = {
+	ID: string;
+	URL: string;
+	caption: string;
+	// Sometimes the title is null, so we need to handle that case.
+	title: string | null;
+	thumbnails: {
+		thumbnail: string;
+	};
+};
+
+/**
+ * WPCOM media list response DTO.
+ */
+type WpcomMediaResponse = {
+	found: number;
+	media: WpcomMediaItem[];
+};
+
+type ConnectedMediaSourceCallback = ( response: WpcomMediaResponse ) => void;
+
+/**
+ * Get media URL for a given MediaSource.
+ *
+ * @param {MediaSource} source - MediaSource to get URL for.
+ * @param {MediaSearch} mediaSearch - MediaCategorySearch to filter for.
+ * @returns {string} Media URL.
+ */
+const getMediaApiUrl = ( source: MediaSource, mediaSearch: MediaSearch ) =>
+	addQueryArgs( `${ WpcomMediaEndpoints.List }${ source }`, {
+		...( mediaSearch.search && { search: mediaSearch.search } ),
+		number: mediaSearch.per_page || 25,
+		path: 'recent',
+	} );
+
+/**
+ * Maps a WPCOM media item to a Gutenberg media item.
+ *
+ * @param {WpcomMediaItem} item - WPCOM media list item to map.
+ * @returns {MediaItem} Mapped media category item.
+ */
+const mapWpcomMediaToMedia = ( item: WpcomMediaItem ): MediaItem => ( {
+	sourceId: item.ID,
+	id: item.ID,
+	caption: item?.caption ?? '',
+	previewUrl: item.thumbnails.thumbnail,
+	title: item?.title ?? '',
+	url: item.URL,
+} );
+
+/**
+ * Builds a Gutenberg media category object.
+ *
+ * @param {string} name - Name of the media category.
+ * @param {string} label - Label of the media category.
+ * @param {string} searchPlaceholder - Search placeholder of the media category.
+ * @param {MediaSource} source - MediaSource of the media category.
+ * @param {MediaSearch} defaultSearch - Default search of the media category.
+ * @returns {object} Media category object.
+ */
+const buildMediaCategory = (
+	name: string,
+	label: string,
+	searchPlaceholder: string,
+	source: MediaSource,
+	defaultSearch: MediaSearch
+) => ( {
+	name: name,
+	labels: {
+		name: label,
+		search_items: searchPlaceholder,
+	},
+	mediaType: 'image',
+	fetch: async ( mediaCategorySearch: MediaSearch ) =>
+		await apiFetch( {
+			path: getMediaApiUrl( source, {
+				...mediaCategorySearch,
+				...defaultSearch,
+			} ),
+			method: 'GET',
+		} )
+			.then( ( response: WpcomMediaResponse ) => response.media.map( mapWpcomMediaToMedia ) )
+			// Null object pattern, we don't want to break if the API fails.
+			.catch( () => [] ),
+	getReportUrl: null,
+	isExternalResource: false,
```

Comment:
Shouldn't this be `true` since the Pexels resources are external?

Reply:
Yes